src/main/java/edu/ksu/canvas/model/CommunicationChannel.java

```diff
@@ -0,0 +1,99 @@
+package edu.ksu.canvas.model;
+
+import com.google.gson.annotations.SerializedName;
+import edu.ksu.canvas.annotation.CanvasField;
+import edu.ksu.canvas.annotation.CanvasObject;
+
+import java.io.Serializable;
+
+/**
+ * Class to represent Canvas user communication channels. See the
+ * <a href="https://canvas.instructure.com/doc/api/communication_channels.html">Communication Channels API</a>
+ * documentation.
+ */
+@CanvasObject(postKey = "communication_channel")
+@SuppressWarnings("serial")
+public class CommunicationChannel extends BaseCanvasModel implements Serializable {
+
+    public enum WorkflowState {
+        @SerializedName("active")
+        ACTIVE,
+        @SerializedName("unconfirmed")
+        UNCONFIRMED;
+
+        @Override
+        public String toString() { return name().toLowerCase(); }
+    }
+
+    public enum Type {
+        @SerializedName("email")
+        EMAIL,
+        @SerializedName("push")
+        PUSH,
+        @SerializedName("sms")
+        SMS,
+        @SerializedName("twitter")
+        TWITTER;
+
+        @Override
+        public String toString() { return name().toLowerCase(); }
+    }
+
+    private String id;
+    private String userId;
```

Comment:
To me this is a bad pattern.
The value that is always returned from Canvas will always be a `Long` and so ideally we should make this a `Long` as well as then you don't have to do any parsing in the application code. However for sending requests to Canvas it is very useful to be able to use alternative IDs: https://canvas.instructure.com/doc/api/file.object_ids.html
That way you can specify users by SIS ID.

I wonder if we should change the API style a little so that the `String` user ID can be passed into the calls but the object just has a `Long`

Reply:
We can leave this issue for now, but continuing to build on suboptimal solutions isn't good.